Common/AnatupleProducer.py

```diff
@@ -12,43 +20,69 @@ def createAnatuple(inFile, outFile, period, sample, X_mass, snapshotOptions):
-    df = df.Define("Bjet1_mass", "HbbCandidate.leg_p4[0].M()")
-    df = df.Define("Bjet1_HHBtagScore", "HbbCandidate.leg_p4[0].M()")
-    df = df.Define("Bjet2_pt", "HbbCandidate.leg_p4[1].Pt()")
-    df = df.Define("Bjet2_eta", "HbbCandidate.leg_p4[1].Eta()")
-    df = df.Define("Bjet2_phi", "HbbCandidate.leg_p4[1].Phi()")
-    df = df.Define("Bjet2_mass", "HbbCandidate.leg_p4[1].M()") 
-    df = df.Define("Bjet2_HHBtagScore", "HbbCandidate.leg_p4[1].M()")  
+    df = Baseline.DefineHbbCand(df)  
+    df = df.Define("channelId","ChannelToHHbTagInput(httCand.channel())")
+    df = df.Define("is_data", "1")
+    df = df.Define("is_TuneCP5", "0")
+    df = df.Define("weight", "1") 
+    df = df.Define("lumi", "luminosityBlock")
+    df = df.Define("evt", "event")
+    deepTauScores= ["rawDeepTau2017v2p1VSe","rawDeepTau2017v2p1VSmu",\
+                "rawDeepTau2017v2p1VSjet", "rawDeepTau2018v2p5VSe", "rawDeepTau2018v2p5VSmu",\
+                "rawDeepTau2018v2p5VSjet"] 
+    colToSave = ["event","lumi","run","sample", "period", "X_mass","channelId", "is_data","is_TuneCP5",\
+                "MET_pt", "MET_phi","PuppiMET_pt", "PuppiMET_phi",\
+                "DeepMETResolutionTune_pt", "DeepMETResolutionTune_phi","DeepMETResponseTune_pt", "DeepMETResponseTune_phi"] 
+ 
+   
+    for leg_idx in [0,1]:
+        df = df.Define(f"tau{leg_idx+1}_pt", f"httCand.leg_p4[{leg_idx}].Pt()")
+        colToSave.append(f"tau{leg_idx+1}_pt")
+        df = df.Define(f"tau{leg_idx+1}_eta", f"httCand.leg_p4[{leg_idx}].Eta()")
+        colToSave.append(f"tau{leg_idx+1}_eta")
+        df = df.Define(f"tau{leg_idx+1}_phi", f"httCand.leg_p4[{leg_idx}].Phi()")
+        colToSave.append(f"tau{leg_idx+1}_phi")
+        df = df.Define(f"tau{leg_idx+1}_mass", f"httCand.leg_p4[{leg_idx}].M()")
+        colToSave.append(f"tau{leg_idx+1}_mass")
+        df = df.Define(f"tau{leg_idx+1}_charge", f"httCand.leg_charge[{leg_idx}]")
+        colToSave.append(f"tau{leg_idx+1}_charge")
+        for deepTauScore in deepTauScores:
+            df = df.Define(f"tau{leg_idx+1}_{deepTauScore}", f"deepTauScore(Tau_{deepTauScore},\
+            httCand.leg_type[{leg_idx}],httCand.leg_index[{leg_idx}])")
+            colToSave.append(f"tau{leg_idx+1}_{deepTauScore}")
+        df = df.Define(f"b{leg_idx+1}_pt", f"HbbCandidate.leg_p4[{leg_idx}].Pt()")
+        colToSave.append(f"b{leg_idx+1}_pt")
+        df = df.Define(f"b{leg_idx+1}_eta", f"HbbCandidate.leg_p4[{leg_idx}].Eta()")
+        colToSave.append(f"b{leg_idx+1}_eta")
+        df = df.Define(f"b{leg_idx+1}_phi", f"HbbCandidate.leg_p4[{leg_idx}].Phi()")
+        colToSave.append(f"b{leg_idx+1}_phi")
+        df = df.Define(f"b{leg_idx+1}_mass", f"HbbCandidate.leg_p4[{leg_idx}].M()")
+        colToSave.append(f"b{leg_idx+1}_mass")
+        df = df.Define(f"b{leg_idx+1}_HHBtagScore", f"HbbCandidate.leg_HHbTagScore[{leg_idx}]")
+        colToSave.append(f"b{leg_idx+1}_HHBtagScore")
+        
```

Comment:
missing
- jet taggers
- hadron and parton flavours

Reply:
added